### PR TITLE
Refactor model methods related to plate merge and division, add automated tests of these functionalities

### PR DESCRIPTION
**Pull request by pjanik**
It's a similar test to the one used by the serialization method.
The idea is to have two models, make a set of actions in one of them, make sure these actions bring it to the initial state, and finally run both models for some time and check if they stay identical.

### CODE REVIEW

**Review by knowuh — APPROVED** [review]
These seem like nice improvements.

I had one small suggestion to add a comment to a test.

File: test/plate-division-merge.test.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+// eslint-disable-next-line jest/expect-expect
+test("plate division and merging consistency", () => {
+  const model1 = new Model(modelImgData, initFunc);
+  
+  // Divide the first plate.
+  const newPlateAdded = model1.dividePlate(model1.plates[0]);
+  expect(newPlateAdded).toEqual(true);
+  expect(model1.plates.length).toEqual(4);
+  expect(model1.plates[1].id).toEqual(3);
```
[the remark this answers is not in the document]
Comment (knowuh) [reply test/plate-division-merge.test.ts:42]: @pjanik  -- this PR looks good, but I just noticed that one of the Travis Jobs failed. 

I am not going to revoke my approval, but I think you should check it out.

**Review by knowuh — APPROVED** [review]
Looks good.

**Comment by pjanik** [comment]
@knowuh, thanks. I've just added a comment. Re failing test - it's random and happens for some time already, but usually, a branch or PR build is okay. :wink: It's a Cypress smoke test that sometimes gets stuck. I'll try to take a look at it during the upcoming cleanup time.

**Pull request merged**